### PR TITLE
[MOIA-29748] Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by m-dobler**
Migrates the existing circleCI workflow to GitHub Actions.

The workflow includes the jobs:
- Compile and run all tests
- Check the formatting of the codebase

The corresponding circleCI workflow is defined [here](https://github.com/moia-dev/scynamo/blob/master/.circleci/config.yml). I am not sure how to go about removing the workflow: If I just delete the config in this PR, the circleCI pipeline would still fail with "No configuration was found in your project". I  guess a circleCI admin would have to stop builds in the [project settings](https://app.circleci.com/settings/project/github/moia-dev/scynamo?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fmoia-dev%2Fscynamo).

### CODE REVIEW

File: .github/workflows/pr-validation.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: PR Validation
+on:
+  push:
+    branches-ignore:
+      - master
```
Comment (m-dobler) [inline .github/workflows/pr-validation.yml:5]: I am not sure if excluding `master` from this workflow is a good idea, does anyone have an opinion?

Comment (saeltz) [reply .github/workflows/pr-validation.yml:5]: We didn't exclude it for `scala-http-client`, so I'd do the same here.

**Comment by saeltz** [comment]
Release in two steps:
First delete the CircleCI file with a direct push to master, then merge this PR and ask @JannikArndt to stop running CircleCI for this project.

File: .github/workflows/pr-validation.yml (hunk 1 above)
Comment (m-dobler) [reply .github/workflows/pr-validation.yml]: Done in 2f75e7f

**Comment by m-dobler** [comment]
So I removed the circleCI config from `master` and merged the commit into this branch, but now this scapegoat job appeared that the PR is waiting for but doesn't seem to be running.

**Comment by JannikArndt** [comment]
My bad, fixed it 👍

**Comment by m-dobler** [comment]
@JannikArndt Thanks 🙏 
Can you explain what was going on and how you fixed it, because I really have no clue 😅

**Comment by JannikArndt** [comment]
Sure! If you go to the Branch-Settings and edit the settings for the `master` branch…
image
…you can select which checks are _required_ to pass to allow merging:
image

There we've had the Circle-checks selected. I changed it to the GitHub Actions, but included the `scapegoat` step. Somehow that doesn't run though. I deselected it (for now) until we actually have it 😊

**Pull request merged**